File: python/src/tasks.py
import math
#task1
def even_or_odd(number):
    if number % 2 == 0:
        return "Even"
    else:
        return "Odd"



#task2
def count_sheeps(sheep):
    return sheep.count(True)


#task3
def monkey_count(n):
    result = []
    for num in range(1, n + 1):
        result.append(num)
    return result



#task4
def paperwork(n, m):
    return n * m if n > 0 and m > 0 else 0 



#task5
def hero(b, d):
  bulletsneeded = d * 2
  if b >= bulletsneeded:
    return True
  else:
    return False


if even_or_odd == "main":
    even_or_odd(23,33,22)
if count_sheeps == "main":
    count_sheeps([True,False,True,False])
if monkey_count == "main":
    monkey_count(33)
if paperwork == "main":
    paperwork(33,12)
if hero == "main":
    hero(12,89)

File: typescript/tasks.ts
import { IntRange } from './types';
import { numsToString, mapping } from './variables';

const tasks = {
  evenOrOdd: (number: number): 'Even' | 'Odd' => {
    return number % 2 == 0 ? 'Even' : 'Odd';
  },
  coutingSheep: (sheeps: boolean[]): number => {
    return sheeps.reduce((acc, sheep) => {
      return sheep ? (acc += 1) : acc;
    }, 0);
  },
  countTheMonkeys: (number: number): number[] => {
    if (number <= 0) {
      return [];
    }

    const result: number[] = [];
    for (let i = 1; i <= number; i++) {
      result.push(i);
    }
    return result;
  },
  greenBottles: (number: IntRange<1, 10>): void => {
    for (let i = number; i > 0; i--) {
      const bottleNumbers =
        numsToString[i - 1][0].toUpperCase() + numsToString[i - 1].slice(1);
      if (i >= 1) {
        console.log(`${bottleNumbers} green bottles hanging on the wall,`);
        console.log(`${bottleNumbers} green bottles hanging on the wall,`);
      }

      if (i == 1) {
        console.log('If that one green bottle should accidentally fall,');
        console.log("There'll be no green bottles hanging on the wall.");
      } else {
        console.log('And if one green bottle should accidentally fall,');
        console.log(
          `There'll be ${
            numsToString[i - 2]
          } green bottles hanging on the wall. \n`
        );
      }
    }
  },
  schoolPaperwork: (n: number, m: number): number => {
    if (n < 0 || m < 0) {
      return 0;
    }
    return n * m;
  },
  defeatingDragons: (bulletsCount: number, dragonsCount: number): boolean => {
    return Math.floor(bulletsCount / 2) >= dragonsCount ? true : false;
  },
  polishAlphabet: (str: string): string => {
    return str.replace(/[^a-zA-Z]/g, (c) => mapping[c] ?? c);
  },
  findAll: (arr: number[], num: number): number[] => {
    return arr.reduce<number[]>(
      (acc, currentValue, index) =>
        num === currentValue ? [...acc, index] : acc,
      []
    );
  },
  sumOfMinimums: (arr: number[][]): number => {
    return arr.reduce((acc, currentArr) => acc + Math.min(...currentArr), 0);
  }
};

export default tasks;
